src/test/java/com/epam/test_generator/services/SuitServiceTest.java

```diff
@@ -369,7 +376,8 @@ public void updateSuitRowNumber_throwsException_DuplicateRowNumber() {
 
     @Test(expected = NotFoundException.class)
     public void getSuitsFromProject_inValidId_NotFound() {
-        when(projectService.getProjectByProjectId(SIMPLE_PROJECT_ID)).thenReturn(expectedProject);
+        ReflectionTestUtils.setField(projectService, "projectDAO", projectDAO);
```

Comment:
Why you need `ReflectionTestUtils` here?

Reply:
it's because we have to mock behavior of projectDAO and inject it into mocked 'projectService.' Otherwise, to satisfy test scenario we need to do something like that - 'when(projectService.getProjectByProjectId(SIMPLE_PROJECT_ID)).thenThrow(NotFoundException.class)'

Reply:
So what's the problem with 
```
when(projectService.getProjectByProjectId(SIMPLE_PROJECT_ID)).thenThrow(NotFoundException.class)
```

Reply:
Correct me if I'm wrong, but I think in this way we don't actually test this scenario, but just declare that we expect that behavior. If you wish I can change it this way.

Reply:
Ok. Understood.